### DISCUSSION TITLE
Updates to "invalid" (terraform) files

### DISCUSSION THREAD

**Discussion by danieljamesscott** [renovatebot/renovate#32010] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

Should renovate work on invalid files? (Or "How can I find out why renovate is not updating my file"?)

I have some templated terraform code which looks like the following snippet:

```
module "{{{ .name }}}" {
  source                                   = "example.com/terraform-local__module/code/aws"
  version                                  = "v1.20.4"
}
```

This file is processed and used to populate another repository. I would like to have renovate process the file and update the version number when I publish a new version of the module. It is working fine for another file in the same repository where there is no template code. I was trying to work out if renovate is not processing the file because it is invalid hcl?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment]
only valid HCL is supported

**Comment by RahulGautamSingh** [comment]
Try custom regex manager
